src/Microsoft.Tye.Core/DockerDetector.cs

```diff
@@ -14,41 +18,107 @@ public class DockerDetector
 
         public static DockerDetector Instance { get; } = new DockerDetector();
 
-        private DockerDetector()
-        {
-            IsDockerInstalled = new Lazy<Task<bool>>(DetectDockerInstalled);
-            IsDockerConnectedToDaemon = new Lazy<Task<bool>>(DetectDockerConnectedToDaemon);
-        }
+        private bool _isUsable { get; }
+        private string? _unusableReason { get; }
 
-        public Lazy<Task<bool>> IsDockerInstalled { get; }
+        public bool IsPodman { get; }
+        public string AspNetUrlsHost { get; }
+        public string? ContainerHost { get; }
 
-        public Lazy<Task<bool>> IsDockerConnectedToDaemon { get; }
-
-        private async Task<bool> DetectDockerInstalled()
+        public bool IsUsable(out string? unusableReason)
         {
-            try
-            {
-                await ProcessUtil.RunAsync("docker", "version", throwOnError: false, cancellationToken: new CancellationTokenSource(Timeout).Token);
-                return true;
-            }
-            catch (Exception)
-            {
-                // Unfortunately, process throws
-                return false;
-            }
+            unusableReason = _unusableReason;
+            return _isUsable;
         }
 
-        private async Task<bool> DetectDockerConnectedToDaemon()
+        private DockerDetector()
         {
+            AspNetUrlsHost = "localhost";
             try
             {
-                var result = await ProcessUtil.RunAsync("docker", "version", throwOnError: false, cancellationToken: new CancellationTokenSource(Timeout).Token);
-                return result.ExitCode == 0;
+                ProcessResult result;
+                try
+                {
+                    // try to use podman.
+                    result = ProcessUtil.RunAsync("podman", "version -f \"{{ .Client.Version }}\"", throwOnError: false, cancellationToken: new CancellationTokenSource(Timeout).Token).Result;
```

Comment:
It's been a while, but is it possible for someone to have both podman and docker locally? If so, having a way to select which one should be used?

Reply:
Yes, both can be installed side-by-side. In most cases, only one will be installed.
Do you want to make it configurable in `tye.yaml`, or in some other way?

Reply:
I think it needs to be configurable.